drivers/scheduler.c

```diff
@@ -294,8 +294,8 @@ scheduler_register_event(scheduler_t *s, char mode, int fd,
 	event->id       = s->uuid++;
 	event->masked   = 0;
 
-	if (!s->uuid)
-		s->uuid++;
+	if (s->uuid < 0)
+		s->uuid = 1;
```

Comment:
Fair point @geosharath, I didn't consider that event_id will have to be changed too. Looks good to me.